### ISSUE TITLE
Discuss: the use of overhead

### ISSUE THREAD

**Issue by harendra-kumar** [vincenthz/hs-gauge#7]
I cannot get my head around this:

```
      stime     = measure (measTime . rescale) .
                  G.filter ((>= threshold) . measTime) . G.map fixTime .
                  G.tail $ meas
      fixTime m = m { measTime = measTime m - overhead / 2 }
```

Why are we fixing the measurement time using the overhead? The overhead is the amount of time `measure` itself takes. How is that relevant? `measTime` has no relation to that. Maybe I am missing something but it looks to me that overhead and its use here can just be removed.

**Comment by vincenthz** [comment]
I don't really have a good understanding here, but the measurement made is all over the place. I'm surprised this get any sort of valid number at the end; The measTime oscillate pretty widely between 1 us to 200ms on a random bench.

**Comment by vincenthz** [comment]
I don't really understand the lowlevel analysis code yet, so it's likely going to take some time to sort out, but I think cleaning up, documenting, refactor should do the job eventually.

**Comment by harendra-kumar** [comment]
This looks outright wrong, it maybe a leftover artifact of some previous method of measurement. If I remove this, results become saner, standard deviation and variance reduces. It is not affecting the results too significantly because we are deducting this overhead from the accumulated time of many iterations and then we are rejecting measurement times which are below a threshold. So these factors are containing the impact of this to a small error.

**Comment by harendra-kumar** [comment]
This computation is used to compute mean and standard deviations and to classify outliers. It is not used for reporting the time in the first line of the results. The mean is in fact wrong because of this. Notice the mean in this result:

```
time                 27.88 ms   (27.05 ms .. 28.92 ms)
                     0.996 R²   (0.992 R² .. 0.999 R²)
mean                 21.70 ms   (19.26 ms .. 23.49 ms)
std dev              3.898 ms   (2.658 ms .. 5.770 ms)
```

The confidence interval of time is from 27.05 to 28.92 and the mean is 21.70 and the stddev is 3.898. When I dump all the measurements time measured is around 27ms for all measurements. How is that possible? When I correct this, here are the results:

```
time                 27.52 ms   (26.57 ms .. 28.69 ms)
                     0.994 R²   (0.986 R² .. 0.998 R²)
mean                 27.24 ms   (26.73 ms .. 27.94 ms)
std dev              1.290 ms   (810.3 μs .. 2.166 ms)
```

Now the mean and the std dev are believable.

**Comment by harendra-kumar** [comment]
Fixed by #21
